trestle/core/rules_validator.py

```diff
@@ -0,0 +1,145 @@
+# -*- mode:python; coding:utf-8 -*-
+
+# Copyright (c) 2023 IBM Corp. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Validate by confirming rule parameter values are consistent."""
+import logging
+
+from trestle.common.common_types import TopLevelOscalModel
+from trestle.common.list_utils import as_list
+from trestle.core.control_interface import ControlInterface
+from trestle.core.validator import Validator
+from trestle.oscal.component import ComponentDefinition
+from trestle.oscal.ssp import SystemSecurityPlan
+
+logger = logging.getLogger(__name__)
+
+
+class RulesValidator(Validator):
+    """Validator to confirm all rule parameter values are consistent."""
+
+    @staticmethod
+    def val_diff_param_values(existing_param: list, shared_param_values: list) -> list:
+        """
+        Test differences between rule param values.
+
+        args:
+            existing_param: Existing rule param values list.
+            shared_param_values: Current rule param values list.
+
+        returns:
+            Difference between values if exist
+        """
+        for param_value in as_list(existing_param):
+            # order params lists and gets differences
+            param_value['values'].sort()
+            shared_param_values.sort()
+            param_vals_set = set(param_value['values'])
+            difference = [x for x in shared_param_values if x not in param_vals_set]
+            # if there´s a difference in params, log error
+            return difference
+
+    def model_is_valid(self, model: TopLevelOscalModel, quiet: bool) -> bool:
```

Comment:
The code in this function is very complex due to the many internal loops and has low reusability. You can split this code into more reusable pieces of code

Reply:
Also quiet is not used. Although it probably shouldn't be used anyways as it would interfere with `-verbose` flag. Use logger levels instead to assign an appropriate level to the message

Reply:
The `quiet` is required because it is in the validator ABC signature and it is a command line option specific to validators.  So it's ok here even if not used.

Reply:
Indeed, forgot we still had this one, thanks for pointing this out